source/parsing/Preprocessor_macros.cpp

```diff
@@ -56,16 +56,29 @@ void Preprocessor::createBuiltInMacro(string_view name, int value, string_view v
 #undef NL
 }
 
-MacroActualArgumentListSyntax* Preprocessor::handleTopLevelMacro(Token directive) {
+std::pair<MacroActualArgumentListSyntax*, Trivia> Preprocessor::handleTopLevelMacro(
+    Token directive) {
+    static Trivia emptyTrivia(TriviaKind::Unknown, string_view(nullptr, 0));
     auto macro = findMacro(directive);
     if (!macro.valid()) {
+        if (options.ignoreDirectives.find(directive.valueText().substr(1)) !=
+            options.ignoreDirectives.end()) {
+            std::vector<Token> ignore_tokens;
+            while (currentToken.kind != TokenKind::EndOfFile && peekSameLine())
+                ignore_tokens.push_back(consume());
+            if (ignore_tokens.empty())
+                return std::make_pair(nullptr, emptyTrivia);
+            else
+                return std::make_pair(nullptr, Trivia(TriviaKind::SkippedTokens, ignore_tokens));
```

Comment:
I missed this on previous review. The ignore_tokens vector is on the stack; once you return you're referring to freed memory. You need to make a copy on the heap in order to refer to it like this.

I would rewrite this to use a SmallVector instead and use its copy() method to make a heap copy via the local allocator. You can find examples of this pattern all over the place in the codebase.